pkg/controllers/vdb/unsandboxsubcluster_reconciler.go

```diff
@@ -0,0 +1,132 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vdb
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	vmeta "github.com/vertica/vertica-kubernetes/pkg/meta"
+	"github.com/vertica/vertica-kubernetes/pkg/names"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// UnsandboxSubclusterReconciler will update sandbox config maps for letting
+// sandbox controller to unsandbox the subclusters
+type UnsandboxSubclusterReconciler struct {
+	Log logr.Logger
+	Vdb *vapi.VerticaDB
+	client.Client
+}
+
+// MakeUnsandboxSubclusterReconciler will build a UnsandboxSubclusterReconciler object
+func MakeUnsandboxSubclusterReconciler(log logr.Logger, vdb *vapi.VerticaDB,
+	cli client.Client) controllers.ReconcileActor {
+	return &UnsandboxSubclusterReconciler{
+		Log:    log.WithName("VdbControllerUnsandboxSubclusterReconciler"),
+		Vdb:    vdb,
+		Client: cli,
+	}
+}
+
+// Reconcile will update sandbox config maps for triggering sandbox controller
+func (r *UnsandboxSubclusterReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	// no-op for ScheduleOnly init policy or enterprise db
+	if r.Vdb.Spec.InitPolicy == vapi.CommunalInitPolicyScheduleOnly || !r.Vdb.IsEON() {
+		return ctrl.Result{}, nil
+	}
+
+	// delete expired sandbox config maps
+	if err := r.deleteExpiredSandboxConfigMaps(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update sandbox config maps for sandboxes that need to be unsandboxed
+	if err := r.updateSandboxConfigMaps(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// updateSandboxConfigMaps will add a trigger ID to sandbox config maps for triggering sandbox controller
+func (r *UnsandboxSubclusterReconciler) updateSandboxConfigMaps(ctx context.Context) error {
+	unsandboxSbScMap := r.Vdb.GenSandboxSubclusterMapForUnsandbox()
+	for sb := range unsandboxSbScMap {
+		nm := names.GenSandboxConfigMapName(r.Vdb, sb)
+		cm := &corev1.ConfigMap{}
+		err := r.Client.Get(ctx, nm, cm)
+		if err != nil {
+			r.Log.Error(err, "failed to retrieve sandbox config map", "configMapName", nm.Name)
+			return err
+		}
+		cm.Annotations[vmeta.SandboxControllerUnsandboxTriggerID] = uuid.NewString()
+		err = r.Client.Update(ctx, cm)
```

Comment:
Can we use the `SandboxConfigMapManager` for this? We do something similar to this to wake up the controller for upgrade.